File: app/src/main/java/com/ksjd/testem/CardHistory.kt
package com.ksjd.testem

enum class HistorySourceType {
    TICKET,
    TRANSACTION
}

data class CardHistoryItem(
    val id: String,
    val sourceType: HistorySourceType,
    val timestampMs: Long,
    val title: String,
    val subtitle: String,
    val amountText: String,
    /** Signed balance change in cents (negative = spent), null if unknown. */
    val amountCents: Long? = null,
    /** Stop where a ticket was bought, if known. */
    val stopName: String = "",
    val isTopUp: Boolean = false
)

data class CardHistoryState(
    val isLoading: Boolean = false,
    val items: List<CardHistoryItem> = emptyList(),
    val errorMessage: String = "",
    val lastUpdatedMs: Long = 0L
)

File: iosApp/iosApp/CardHistory.swift
import Foundation

enum HistorySourceType: String {
    case ticket
    case transaction
}

struct CardHistoryItem: Identifiable {
    let id: String
    let sourceType: HistorySourceType
    let timestampMs: Int64
    let title: String
    let subtitle: String
    let amountText: String
}

struct CardHistoryState {
    var isLoading = false
    var items: [CardHistoryItem] = []
    var errorMessage = ""
    var lastUpdated: Date?
}
